lib/pages/edit_task.dart

```diff
@@ -3,65 +3,49 @@ import 'package:go_router/go_router.dart';
 import 'package:task_list_app/constants/constants.dart';
 import 'package:task_list_app/models/task.dart';
 
-class EditTask extends StatefulWidget {
+class EditTask extends StatelessWidget {
   final Task task;
-  final Function(Task) onTaskUpdated;
-  const EditTask({super.key, required this.task, required this.onTaskUpdated});
-
-  @override
-  State<EditTask> createState() => _EditTaskState();
-}
+  final Function(String) onTaskUpdated;
 
-class _EditTaskState extends State<EditTask> {
-  late TextEditingController _controller;
-  @override
-  void initState() {
-    super.initState();
-    _controller = TextEditingController(text: widget.task.name);
-  }
+  const EditTask({super.key, required this.task, required this.onTaskUpdated});
 
   @override
   Widget build(BuildContext context) {
+    TextEditingController controller = TextEditingController(text: task.name);
```

Comment:
I liked the name for your controller above called `addTask`. `addTask` describes what it does rather than what it is.

Reply:
done